components/UI/Drawer/Drawer.tsx

```diff
@@ -113,107 +156,135 @@ const Drawer = () => {
         role="presentation"
         onKeyDown={(e) => toggler(e)}
       >
-        <div className={styles.content}>
-          <h3 style={{ maxWidth: "45ch" }}>{formatted_address}</h3>
-          <p>{formattedCoordinates}</p>
-          <div className={styles.contentButtons}>
-            {mapsButtons.map((button) => (
-              <Button
-                key={button.label}
-                variant="contained"
-                onClick={() => {
-                  button.urlCallback(
-                    geometry.location.lat,
-                    geometry.location.lng
-                  );
-                }}
-                color={button.color}
-                className={styles.externalLinkButton}
-                startIcon={button.icon}
-              >
-                {button.label}
-              </Button>
-            ))}
-          </div>
-          <div>
-            <TextField
-              fullWidth
-              variant="standard"
-              size="small"
-              value={generateGoogleMapsUrl(
-                geometry.location.lat,
-                geometry.location.lng
-              )}
-              InputProps={{
-                readOnly: true,
-              }}
-            />
-            <div className={styles.actionButtons}>
-              <Button
-                variant="outlined"
-                className={styles.clipboard}
-                size="small"
-                fullWidth
-                onClick={() =>
-                  copyBillboard(
-                    `https://www.google.com/maps/@${geometry.location.lat.toString()},${geometry.location.lng.toString()},22z`
-                  )
-                }
-                startIcon={<CopyAll className={styles.btnIcon} />}
-              >
-                Kopyala
-              </Button>
-              <Button
-                variant="outlined"
-                className={styles.clipboard}
+        {isLoading && (
+          <Box
+            sx={{
+              height: "300px",
+              display: "flex",
+              justifyContent: "center",
+              alignItems: "center",
+            }}
+          >
+            <CircularProgress />
+          </Box>
+        )}
+        {error && (
+          <Box
+            sx={{
+              height: "300px",
```

Comment:
Buraya fixed bir height vermeden cozmenin bir yolu var mi? 

(Bunu vermezsek asagidan cilar drawer loading indicator heighti kadar oluyor)

Reply:
Error component'ini component/Error altinda generic hale getirsek daha kullanisli olabilir.

Reply:
Fixed height yerine responsive min height ayarlamak daha esnek olur.

Reply:
SkeletonLoader koyabiliriz ama onlara da fixed bir height vermek gerekir.

Reply:
Aslinda bu error componenti cok genericmis. Butun return kismi suna donusse aslinda cok guzel olur bence:

```jsx
if (error) throw error;
if (isLoading) return <Spinner />
if (!data) return <Empty />

return (...)
```

error'i da kullanilan yerde CatchBoundary ile yakalayip error componentini orada minHeight ile istedigimiz gibi kullanabiliriz.

Reply:
@buraksaraloglu Katiliyorum, bence butun Drawer'in refactor edilmesi lazim. Baska bir PR'da yapsak olur mu?